### PR TITLE
Refactor vision benchmark

### LINKED ISSUE #769: Refactor vision and llm benchmarks
We should refactor vision and llm benchmarks similar to encoder benchmark in #748:
Every model should define and pass to benchmark:
1. `load_inputs` function
2. `preprocess_input` function
3. `process_output` function

We should also refactor these functions in [tt_forge_models](https://github.com/tenstorrent/tt-forge-models) and use them as defaults.

Also, we should remove [cpu_measurment](https://github.com/tenstorrent/tt-forge/pull/748#discussion_r2646852480) and [training parameter](https://github.com/tenstorrent/tt-forge/pull/748#discussion_r2646851776) from benchmarks.

### PR DESCRIPTION
**Pull request by vkovacevicTT**
Closes #769 

### What's changed

For all benchmarks:
- removed training and cpu measurement logic

In vision models benchmark:
- every model defines two functions:
  1. `load_inputs` - loads and preprocesses input batch; now easier to utilize tt_forge_models function
  2. `extract_output_tensor` - gets tensor from result (e.g. logits)
- moved PCC check after print benchmark result for easier debugging
- ~~batched input is loaded every loop instead `loop_count*[batched_input]` before benchmarking~~
- ~~sync is forced on end of every loop to measure full device execution~~ torch_xla.sync() doesn't wait for full device execution
- full benchmark time is measured
- renamed vision_models.py to test_vision.py, llms.py to test_llms.py, encoders.py to test_encoders.py
- removed ufld (Ultra-Fast-Lane-Detection-v1) since it is essentially the same as ufld_v2 for which we have a target

### CODE REVIEW

**Comment by vkovacevicTT** [comment]
@rpavlovicTT 
> I guess you checked performance with this change?

Yes, but this change introduces a weird behaviour - FPS skyrockets.
Resnet: https://github.com/tenstorrent/tt-forge/actions/runs/21008092883/job/60395665321

I am looking into this.

Seems to happen when I generate input_tensor per loop instead of generating a large loop_count*[input_tensor] python list beforehand.
In resnet example we use the same input_tensor in each iteration, but this happens even if we generate different rand tensor every iteration.

**Review by nsumrakTT — APPROVED** [review]

File: benchmark/tt-xla/vision_models.py [hunk 1]
```diff
@@ -16,50 +17,50 @@
...
 DEFAULT_DATA_FORMAT = "bfloat16"
-DEFAULT_MEASURE_CPU = False
 DEFAULT_EXPERIMENTAL_COMPILE = True
 DEFAULT_REQUIRED_PCC = 0.97
-DEFAULT_READ_LOGITS_FN = lambda output: output
+
+
+def get_dtype(data_format):
+    """Convert data_format string to torch dtype."""
+    return torch.bfloat16 if data_format == "bfloat16" else torch.float32
```
Comment (odjuricicTT) [inline benchmark/tt-xla/vision_models.py]: It's not safe to have `else torch.float32, there could be other formats besides these two.

You can do `elif` for f32 and raise an error in the end if no string is matched.

File: benchmark/tt-xla/vision_models.py [hunk 2]
```diff
@@ -112,136 +111,385 @@ def test_vision(
...
         required_pcc=0.90,
-        read_logits_fn=read_logits_fn,
     )
 
 
 def test_segformer(output_file):
     from third_party.tt_forge_models.segformer.semantic_segmentation.pytorch.loader import ModelLoader, ModelVariant
 
+    # Configuration
+    data_format = "bfloat16"
```
Comment (odjuricicTT) [inline benchmark/tt-xla/vision_models.py]: Do we need this as string? Just use torch dtypes here as well?
